proto/src/main/proto/grpc.proto

```diff
@@ -598,6 +598,150 @@ message AddressBalanceInfo {
     bool is_address_unused = 4;
 }
 
+///////////////////////////////////////////////////////////////////////////////////////////
+// MoneroConnections
+///////////////////////////////////////////////////////////////////////////////////////////
+
+service MoneroConnections {
+    rpc AddConnection (AddConnectionRequest) returns (AddConnectionResponse) {
+    }
+
+    rpc RemoveConnection(RemoveConnectionRequest) returns (RemoveConnectionResponse) {
+    }
+
+    rpc GetConnection(GetConnectionRequest) returns (GetConnectionResponse) {
+    }
+
+    rpc GetConnections(GetConnectionsRequest) returns (GetConnectionsResponse) {
+    }
+
+    rpc SetConnection(SetConnectionRequest) returns (SetConnectionResponse) {
+    }
+
+    rpc ExtendedSetConnection(ExtendedSetConnectionRequest) returns (ExtendedSetConnectionResponse) {
```

Comment:
This should be removed, only using SetConnection where SetConnectionRequest contains all fields that can be set. Login credentials are only overwritten if provided.

Reply:
Login credentials are always overwritten, if `MoneroConnectionManager.setConnection(MoneroRpcConnection)` is called.  So the `SetConnection` endpoint can be used as switch, and `ExtendedSetConnection` as update and switch. If these are merged, one cannot switch to a stored password-protected connection, as the password will get overwritten with `null`.

Reply:
Maybe we should differentiate between update and switch and provide a `UpdateConnection(UriConnection)` and `SwitchConnection(string)` endpoint?